packer/README.md

```diff
@@ -34,6 +34,58 @@ vagrant box remove centos7-katello-devel-stable
 sudo virsh vol-delete --pool default centos7-katello-devel-stable_vagrant_box_image_0.img
 ```
 
+### Creating version specific dev boxes
+
+To create a version specific dev box, simply:
+
+```
+cd ./forklift/packer
+```
+
+and run this comand (replacing 3.18 with the correct version):
+
+```
+./scripts/build_stable_dev_box.rb 3.18
+```
+
+To test the newly created box:
+1.  add the printed box definition to vagrant/boxes.d/99-local.yaml 
+2.  run printed 'vagrant box add' command
+3.  vagrant up centos7-katello-3.18-stable
+4.  Verify that 'rpm -q katello-repos' is correct and the foreman and katello git repos are on the correct branches
+
+To upload the box:
+1. While you can try with the latest version of vagrant, versions 2.2.11-2.2.14 cannot properly upload boxes.  Download 2.2.9 (this will not interfere with your rpm installed vagrant):
+```
+wget https://releases.hashicorp.com/vagrant/2.2.9/vagrant_2.2.9_linux_amd64.zip
+unzip vagrant_2.2.9_linux_amd64.zip
+sudo mv vagrant /usr/bin/vagrant-upstream
```

Comment:
Having a lot of trouble with this:

```
[jomitsch@jomitsch tmp]$ sudo mv vagrant /usr/bin/vagrant-upstream
[jomitsch@jomitsch tmp]$ vagrant-upstream -h
Vagrant failed to initialize at a very early stage:

The plugins failed to initialize correctly. This may be due to manual
modifications made within the Vagrant home directory. Vagrant can
attempt to automatically correct this issue by running:

  vagrant plugin repair

If Vagrant was recently updated, this error may be due to incompatible
versions of dependencies. To fix this problem please remove and re-install
all plugins. Vagrant can attempt to do this automatically by running:

  vagrant plugin expunge --reinstall

Or you may want to try updating the installed plugins to their latest
versions:

  vagrant plugin update

Error message given during initialization: Unable to resolve dependency: user requested 'vagrant-libvirt (= 0.0.45)'
[jomitsch@jomitsch tmp]$ vagrant-upstream plugin install vagrant-libvirt
Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
Fetching formatador-0.2.5.gem
Fetching fog-core-2.2.3.gem
Fetching fog-json-1.2.0.gem
Fetching mini_portile2-2.4.0.gem
Fetching nokogiri-1.10.10.gem
Building native extensions. This could take a while...
Fetching fog-xml-0.1.3.gem
Fetching ruby-libvirt-0.7.1.gem
Building native extensions. This could take a while...
Vagrant failed to properly resolve required dependencies. These
errors can commonly be caused by misconfigured plugin installations
or transient network issues. The reported error is:

ERROR: Failed to build gem native extension.

    current directory: /home/jomitsch/.vagrant.d/gems/2.6.2/gems/ruby-libvirt-0.7.1/ext/libvirt
/tmp/.mount_vagran7xsJaF/usr/bin/ruby2.6 -I /tmp/.mount_vagran7xsJaF/usr/lib/ruby/2.6.0 -r ./siteconf20201222-21704-ymajsy.rb extconf.rb
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/tmp/.mount_vagran7xsJaF/usr/bin/$(RUBY_BASE_NAME)2.6
	--with-libvirt-include
	--without-libvirt-include
	--with-libvirt-lib
	--without-libvirt-lib
	--with-libvirt-config
	--without-libvirt-config
	--with-pkg-config
	--without-pkg-config
extconf.rb:73:in `<main>': libvirt library not found in default locations (RuntimeError)

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /home/jomitsch/.vagrant.d/gems/2.6.2/extensions/x86_64-linux/2.6.0/ruby-libvirt-0.7.1/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /home/jomitsch/.vagrant.d/gems/2.6.2/gems/ruby-libvirt-0.7.1 for inspection.
Results logged to /home/jomitsch/.vagrant.d/gems/2.6.2/extensions/x86_64-linux/2.6.0/ruby-libvirt-0.7.1/gem_make.out

```

Reply:
there's no need to install a plugin with that (in fact i'd recommend against it).  I'm not sure why vagrant-upstream -h  didn't work for you, it worked fine on my centos box and my fedora box.